src/framework/net/httplogin.cpp

```diff
@@ -136,6 +142,63 @@ void LoginHttp::httpLogin(const std::string& host, const std::string& path,
             });
         }
     });
+#else
+    g_asyncDispatcher.detach_task(
+        [this, host, path, port, email, password, request_id, httpLogin] {
+        emscripten_fetch_attr_t attr;
+        emscripten_fetch_attr_init(&attr);
+        strcpy(attr.requestMethod, "POST");
+        static const char* const headers[] = {
+            "Content-Type", "application/json; charset=utf-8",
+            0,
+        };
+        attr.requestHeaders = headers;
+        attr.attributes = EMSCRIPTEN_FETCH_LOAD_TO_MEMORY | EMSCRIPTEN_FETCH_SYNCHRONOUS;
+        std::string body = "{\"email\":\"" + email + "\",\"password\":\"" + password + "\",\"stayloggedin\":true,\"type\":\"login\"}";
+        attr.requestData = body.c_str();
+        attr.requestDataSize = strlen(attr.requestData);
```

Comment:
```suggestion
        attr.requestData = body.data();
        attr.requestDataSize = body.length();
```
in this case it doesn't really matter because we're not dealing with binary data, but if you did the c_str()+strlen approach and the body contained binary data, you would be looking at data corruption.